lib/internal/render.js

```diff
@@ -47,7 +48,7 @@ export function updateInstanceFromProps(instance, props, oldProps = {}) {
     }
     if (listenerRegex.test(key)) {
       const listener = props[key];
-      const type = key.replace(listenerRegex, '$1').toLowerCase();
+      const type = key.replace(listenerRegex, '$1').toLowerCase().replace(listenerColonRegex, ":");
```

Comment:
Since the issue is specific to the `change:foo` type events (in a limited set of bundlers), I think this change could be more specific.  As this is written, it would break any other events that might use the underscore (e.g. `onFoo_bar` would be misinterpreted as `foo:bar`, blocking the ability to make use of a `foo_bar` event).

Instead, I think it would be clearer if we had special handling for `onChange-` prefixed props.

E.g.
```js
  for (let key in props) {
    if (reservedProps[key]) {
      continue;
    }
    // special handling for bundlers that don't work with : in a prop
    if (key.startsWith('onChange-')) {
      key = 'onChange:';
    }
    if (listenerRegex.test(key)) {
      // ...
    }
  }
```

Would a dash (`-`) work in the bundlers that you use, @vincent-lecrubier-skydio?  Minor nit, but I find `onChange-foo` a bit more appealing than `onChange_foo`.

Reply:
I updated the PR with something similar. I still think it makes sense to have the `onChange-` => `onChange:` substitution behavior be nested inside of the `if(listenerRegex.test(key))` block, to only run the regex substitution behavior on listener props (not on all props) which should have a decent performance benefit.